tmt/steps/provision/bootc.py

```diff
@@ -0,0 +1,232 @@
+
+    .. code-block:: yaml
+
+        provision:
+            how: bootc
+            add_deps: false
+            containerimage: localhost/my-image-with-deps
+
+    This plugin is an extension of the virtual.testcloud plugin.
+    Essentially, it takes a container image as input, builds a
+    bootc disk image from the container image, then uses the virtual.testcloud
+    plugin to create a virtual machine using the bootc disk image.
+
+    The bootc disk creation requires running podman as root, this is typically
+    done by running the command in a rootful podman-machine. The podman-machine
+    also needs access to ``/var/tmp/tmt``. An example command to initialize the
+    machine:
+
+    .. code-block:: shell
+
+        podman machine init --rootful --disk-size 200 --memory 8192 \
+        --cpus 8 -v /var/tmp/tmt:/var/tmp/tmt -v $HOME:$HOME
+    """
+
+    _data_class = BootcData
+    _guest_class = GuestTestcloud
+    _guest = None
+    _id = str(uuid.uuid4())[:8]
+
+    # build a "modified" container image from the base image with tmt dependencies added
+    def _build_derived_image(self, base_image: str) -> str:
+        self._logger.debug("Building modified container image with necessary tmt packages/config")
+        containerfile_template = Template('''
+            FROM $base_image
+
+            RUN \
+            dnf -y install cloud-init rsync && \
+            ln -s ../cloud-init.target /usr/lib/systemd/system/default.target.wants && \
+            rm /usr/local -rf && ln -sr /var/usrlocal /usr/local && mkdir -p /var/usrlocal/bin && \
+            dnf clean all
+        ''')
+        containerfile = containerfile_template.substitute({"base_image": base_image})
+        with open(f'{self.workdir}/Containerfile', 'w') as file:
+            file.write(containerfile)
+
+        image_tag = f'localhost/tmtmodified-{self._id}'
+        tmt.utils.Command(
+            "podman", "build", f'{self.workdir}',
+            "-f", f'{self.workdir}/Containerfile',
+            "-t", image_tag
+            ).run(cwd=self.workdir, stream_output=True, logger=self._logger)
+
+        return image_tag
+
+    # build the "base" or user supplied container image
+    def _build_base_image(self, containerfile: str, workdir: str) -> str:
+        image_tag = f'localhost/tmtbase-{self._id}'
+        self._logger.debug("Building container image")
+        tmt.utils.Command(
+            "podman", "build", workdir,
+            "-f", containerfile,
+            "-t", image_tag
+            ).run(cwd=self.workdir, stream_output=True, logger=self._logger)
+        return image_tag
+
+    # build the bootc disk
+    def _build_bootc_disk(self, containerimage: str, image_builder: str) -> None:
+        self._logger.debug("Building bootc disk image")
+        tmt.utils.Command(
+            "podman", "run", "--rm", "--privileged",
+            "-v", "/var/lib/containers/storage:/var/lib/containers/storage",
```

Comment:
Moved the string to a constant at the top of the module. This is the default location for podman container storage. It can be changed in podman configs. I'm not sure it's worth adding an option to change it until we find it necessary. If you think we should add the option up front that works too.

Reply:
Hm, I'm not familiar enough with `podman` & co. to tell how often one might need to change this path. I suppose it's safe for now to be marked as a "constant", we can always add an option later if it becomes clear there's a use case for it.